server/apps/publish/publish_filters/publish_filter.py

```diff
@@ -70,8 +69,46 @@ def create(self, docs, **kwargs):
                     raise SuperdeskApiError.badRequestError('Article not found!')
                 doc['matches'] = self.does_match(doc, article)
             return [doc['matches'] for doc in docs]
+        self._validate_publish_filters(docs)
         return super().create(docs, **kwargs)
 
+    def update(self, id, updates, original):
+        publish_filter = dict(original)
+        publish_filter.update(updates)
+        self._validate_no_circular_reference(publish_filter, publish_filter['_id'])
+        super().update(id, updates, original)
+
+    def delete(self, lookup):
+        referenced_filters = self._get_referenced_publish_filters(lookup.get('_id'))
+        if referenced_filters.count() > 0:
+            references = ''.join([pf['name'] for pf in referenced_filters])
+            raise SuperdeskApiError.badRequestError('Publish filter has been referenced in {}'.format(references))
+        return super().delete(lookup)
+
+    def _validate_publish_filters(self, docs):
+        for doc in docs:
+            if not doc.get('name', None):
+                raise SuperdeskApiError.badRequestError('Name cannot be empty')
```

Comment:
True, since I use the post for testing against an article, I removed it from schema definition. I may need to create a new end-point for further tests (against production stories) then I'll put the name back to the schema.